src/qfvm/statevector.hpp

```diff
@@ -1399,3 +1407,39 @@ void StateVector<real_t>::apply_reset(vector<pos_t> const& qbits) {
   const auto meas = sample_measure_probs(qbits);
   update(qbits, 0, meas.first, meas.second);
 }
+
+
+ //sample results
+template <class real_t>
+std::unordered_map<std::string, int> StateVector<real_t>::measure_samples(vector<std::pair<uint, uint>> meas, int shots){
```

Comment:
better to use `const std::vector<std::pair<int, int>> &` for non-modifiable complex data types